packages/button/src/styledComponents/Button.js

```diff
@@ -80,8 +80,8 @@ function paddingStyle(props) {
   `;
 }
 
-function neutralStyle(props) {
-  const { textColor, iconColor, textHoverColor, color } = colorTypes[NEUTRAL];
+function neutralStyle(props, theme) {
```

Comment:
theme on a component child of ThemeProvider is already present in the props, I think we could keep just one argument here with a little tweak below on the place this function is called